src/server/scripts/Shadowlands/SepulcherOfTheFirstOnes/boss_anduin_wrynn.cpp

```diff
@@ -0,0 +1,4307 @@
+            soul->GetAI()->DoAction(ACTION_DOUBT_GONE);
+    }
+
+    void UpdateAI(uint32 diff) override
+    {
+        UpdateVictim();
+
+        _events.Update(diff);
+
+        while (uint32 eventId = _events.ExecuteEvent())
+        {
+            switch (eventId)
+            {
+                case EVENT_ANDUIN_SOUL:
+                {
+                    // @TODO: Drop stuff if we found a way which covers stun/deathgrip
+                    me->GetMotionMaster()->MovePoint(POINT_ANDUIN_SOUL, AnduinsSoulSpawnPosition);
+                    _events.Repeat(200ms);
+                    break;
+                }
+                default:
+                    break;
+            }
+        }
+    }
+
+private:
+    InstanceScript* _instance;
+    EventMap _events;
+};
+
+// Anduin's Hope - 184493
+struct npc_anduin_wrynn_anduin_hope : public ScriptedAI
+{
+    npc_anduin_wrynn_anduin_hope(Creature* creature) : ScriptedAI(creature),
+        _instance(creature->GetInstanceScript()) { }
+
+    void Reset() override
+    {
+        me->SetRegenerateHealth(false);
+        me->SetHealth(1);
+        me->SetReactState(REACT_PASSIVE);
+    }
+
+    void JustAppeared() override
+    {
+        _instance->SendEncounterUnit(ENCOUNTER_FRAME_ENGAGE, me, 255);
+        me->SetWalk(true);
+        DoCastSelf(SPELL_ANDUIN_SLOW);
+        if (IsHeroic() || IsMythic())
+            DoCastSelf(SPELL_GLOOM);
+
+        _scheduler.Schedule(2s, [this](TaskContext /*task*/)
+        {
+            Position exitPlatform = me->GetFirstCollisionPosition(100.0f, 0);
+            me->GetMotionMaster()->MovePoint(POINT_ESCAPE_PLATFORM, exitPlatform, false, me->GetOrientation());
+        });
+        me->SetHealth(1);
+
+        _scheduler.Schedule(40s, [this](TaskContext /*task*/)
+        {
+            me->DespawnOrUnsummon();
+        });
+    }
+
+    void UpdateAI(uint32 diff) override
+    {
+        UpdateVictim();
+
+        _scheduler.Update(diff);
+    }
+
+    void HealReceived(Unit* /*healer*/, uint32& heal) override
+    {
+        if (me->HealthAbovePctHealed(100, heal))
+        {
+            _instance->SendEncounterUnit(ENCOUNTER_FRAME_DISENGAGE, me);
+            DoCastSelf(SPELL_WILLPOWER_ENERGIZE_SMALL);
+            DoCastSelf(SPELL_SOUL_DESPAWN);
+
+            if (Creature* soul = me->GetInstanceScript()->GetCreature(DATA_ANDUIN_SOUL))
+                soul->GetAI()->DoAction(ACTION_HOPE_RESTORED);
+        }
+    }
+
+    void MovementInform(uint32 type, uint32 pointId) override
+    {
+        if (type != POINT_MOTION_TYPE)
+            return;
+
+        switch (pointId)
+        {
+            case POINT_ESCAPE_PLATFORM:
+            {
+                DoCastSelf(SPELL_SOUL_DESPAWN);
+                break;
+            }
+            default:
+                break;
+        }
+    }
+
+private:
+    InstanceScript* _instance;
+    TaskScheduler _scheduler;
+};
+
+// Fiendish Soul - 183669
```

Comment:
all of the comments naming entryNames and entryIds should be set like entryId - entryName, this applies to ALL comments.